services/graphql/src/routes/reporting.js

```diff
@@ -0,0 +1,93 @@
+const { Parser } = require('json2csv');
+const { ObjectId } = require('mongodb');
+const Reporting = require('../services/reporting');
+const {
+  publisher, deployment, adunit, advertiser,
+  order, lineitem, ad,
+} = require('../mongoose/models');
+
+const { log } = console;
+const { isArray } = Array;
+
+const asyncRoute = fn => (req, res, next) => Promise.resolve(fn(req, res, next)).catch(next);
+const retrieve = (model, $in) => model.find({ _id: { $in } }, ['name', 'size']);
+const getModelValue = (field, row, models) => {
+  try {
+    const [model, subfield] = field.split('.');
+    const found = models[`${model}s`].filter(({ _id }) => `${_id}` === `${row[`${model}Id`]}`);
```

Comment:
Adding the `s` to the model name seems "sloppy". Can this be refined? It assumes a lot about how the models are named.